### PR TITLE
Create text_helper.php

### PR DESCRIPTION
**Pull request by Portaflex**
Added type hints.
Pending tests and documentation ...

### CODE REVIEW

File: CI4/system/Helpers/text_helper.php [hunk 1]
```diff
@@ -0,0 +1,848 @@
+<?php
+use Zend\I18n\Validator\Int;
+use PhpParser\Node\Expr\Cast\String_;
```
Comment (lonnieezell) [inline CI4/system/Helpers/text_helper.php]: Pretty sure we don't need either of these to make the helpers work...

File: CI4/tests/system/Helpers/TextHelperTest.php [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+    }
+    
+    // -----------------------------------------------------------------------
+    
+    public function test_excerpt()
+    {
+        $string = $this->_long_string;
+        $phrase = 'write';
+        $this->assertEquals(excerpt($string, $phrase), '... '.$this->_long_string.' ...');
+    }
```
Comment (lonnieezell) [inline CI4/tests/system/Helpers/TextHelperTest.php]: Could use an additional test to ensure that it's limiting the results correctly within a radius (and preferably at word breaks?) Currently, you're only testing that it's appending an ellipsis the on the beginning and end of the string.

**Comment by lonnieezell** [comment]
Starting to look pretty good! Thanks. If you can take care of those couple of things I've marked, and then get code signing working we'll be good to go. If you're having any troubles with code signing say something and we'll try to figure it out.

**Comment by Portaflex** [comment]
Those two thing fixed, but code signing not. Please help!!

**Comment by lonnieezell** [comment]
No problem. So - out of the steps listed [on this page](https://bcit-ci.github.io/CodeIgniter4/contributing/signing.html#secure-signing) under Secure Signing, what have you managed to do and where are you running into problems? 

I found that when I went through the process, it was easiest to ignore the link titled **git tools**, though that explains the basics. If I followed the links in the steps below that, though, it went pretty smoothly. Those are GitHub help pages and you can choose your operating system at the top of the page to get step-by-step process specific for your OS.

**Comment by Portaflex** [comment]
Changed to another pull request.

**Pull request closed without merging**